### PR TITLE
fastfetch: 2.11.0 -> 2.11.3

### LINKED ISSUE #308849: [Fastfetch] Replace hard coded paths
### Describe the bug

[Fastfetch](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/fa/fastfetch/package.nix) uses `hwdata/pci.ids` and `libdrm/amdgpu.ids` to convert PCI device ID to human-readable product name.

Currently their paths are hard coded as `/usr/share/hwdata/pci.ids` and `/usr/share/libdrm/amdgpu.ids`. It won't work with NixOS of course.

https://github.com/fastfetch-cli/fastfetch/blob/dev/src/detection/gpu/gpu_linux.c#L126-L137

I found some packages that have similar issues. They are patched directly before building: https://github.com/NixOS/nixpkgs/blob/662ae2a0cf412fec9f7ffefa16dc4cb70343814b/pkgs/tools/graphics/mangohud/hardcode-dependencies.patch#L38

Fastfetch needs to be patched too.

### Steps To Reproduce

N/A

### Expected behavior

Fastfetch can properly find `pci.ids` and `amdgpu.ids`

### Screenshots

N/A

### Additional context

Reference: https://github.com/fastfetch-cli/fastfetch/issues/855

Fastfetch provided cmake config option [`CUSTOM_PCI_IDS_PATH`](https://github.com/fastfetch-cli/fastfetch/blob/dev/CMakeLists.txt#L81) for `pci.ids` but not for `amdgpu.ids`. I can add one if needed.

### Notify maintainers

@chewblacka @r-ryantm

### Metadata

N/A

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by chewblacka**
## Description of changes
- 2.11.0 -> 2.11.3
https://github.com/fastfetch-cli/fastfetch/blob/2.11.3/CHANGELOG.md

- set correct nix-store paths for the files `pci.ids` and  `amdgpu.ids`

- Closes https://github.com/NixOS/nixpkgs/issues/308849

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 309042` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
  </ul>
</details>

**Review by khaneliman — CHANGES_REQUESTED** [review]
libdrm hide behind linux check

**Comment by Sigmanificient** [comment]
Result of `nixpkgs-review pr 309042` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
  </ul>
</details>

File: pkgs/by-name/fa/fastfetch/package.nix [hunk 1]
```diff
@@ -67,6 +68,7 @@ stdenv'.mkDerivation (finalAttrs: {
   buildInputs = [
     chafa
     imagemagick_light
+    libdrm
```
Comment (khaneliman) [inline pkgs/by-name/fa/fastfetch/package.nix:71]: move below to stdenv.isLinux

File: pkgs/by-name/fa/fastfetch/package.nix [hunk 2]
```diff
@@ -118,6 +120,8 @@ stdenv'.mkDerivation (finalAttrs: {
 
   cmakeFlags = [
     (lib.cmakeOptionType "filepath" "CMAKE_INSTALL_SYSCONFDIR" "${placeholder "out"}/etc")
+    (lib.cmakeOptionType "filepath" "CUSTOM_PCI_IDS_PATH" "${hwdata}/share/hwdata/pci.ids")
+    (lib.cmakeOptionType "filepath" "CUSTOM_AMDGPU_IDS_PATH" "${libdrm}/share/libdrm/amdgpu.ids")
```
Comment (khaneliman) [inline pkgs/by-name/fa/fastfetch/package.nix:124]: (!stdenv.isDarwin) check

File: pkgs/by-name/fa/fastfetch/package.nix [hunk 3]
```diff
@@ -118,6 +120,8 @@ stdenv'.mkDerivation (finalAttrs: {
 
   cmakeFlags = [
     (lib.cmakeOptionType "filepath" "CMAKE_INSTALL_SYSCONFDIR" "${placeholder "out"}/etc")
+    (lib.cmakeOptionType "filepath" "CUSTOM_PCI_IDS_PATH" "${hwdata}/share/hwdata/pci.ids")
```
Comment (khaneliman) [inline pkgs/by-name/fa/fastfetch/package.nix:123]: looks like we're only including hwdata in build inputs for linux atm, can move that to the linux section with libdrm too

Comment (chewblacka) [reply pkgs/by-name/fa/fastfetch/package.nix]: Many thanks for the review. Have amended with the requested changes. I think that is correct now.

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 309042` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 309042` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
Kicked off Linux build before leaving so couldn't check if it actually fixed my gpu detection. But it builds 😀

**Comment by Gerg-L** [comment]
Builds and runs on x86_64-linux

before:
```
GPU 1: NVIDIA 
GPU 2: AMD  [Integrated]
```
after:
```
GPU 1: NVIDIA GeForce RTX 3060 Lite Hash Rate [Discrete]
GPU 2: AMD Raphael @ 0.60 GHz [Integrated]
```

LGTM

**Review by Gerg-L — APPROVED** [review]

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 309042` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 309042` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
> 
> after:
> 
> ```
> 
> GPU 1: NVIDIA GeForce RTX 3060 Lite Hash Rate [Discrete]
> 
> GPU 2: AMD Raphael @ 0.60 GHz [Integrated]
> 
> ```
> 

Excited for the GPU detection to work again. Thanks :)

**Pull request merged**